### ISSUE TITLE
BUG: Polygons with holes are not drawn correctly

### ISSUE THREAD

**Issue by CNFeffery** [geopandas/geopandas#3353]
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of geopandas.

- [ ] (optional) I have confirmed this bug exists on the main branch of geopandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example

```python
import geopandas as gpd
from shapely.geometry import Polygon

demo_gdf = gpd.GeoDataFrame(
    geometry=[
        Polygon(
            [(0, 0), (0, 5), (5, 5), (5, 0)],
            [
                [(1, 1), (1, 2), (2, 2), (2, 1)],
                [(3, 2), (3, 3), (4, 3), (4, 2)],
            ],
        )
    ]
)
demo_gdf.plot(edgecolor='black');
```

#### Problem description

Polygons with holes are not drawn correctly.
![image](https://github.com/geopandas/geopandas/assets/49147660/9ed18ea6-edf9-473f-8e66-6f4a72c57b23)

#### Output of ``geopandas.show_versions()``

<details>

SYSTEM INFO
-----------
python     : 3.9.19 | packaged by conda-forge | (main, Mar 20 2024, 12:38:46) [MSC v.1929 64 bit (AMD64)]
executable : C:\miniforge\envs\data-science\python.exe
machine    : Windows-10-10.0.22631-SP0

GEOS, GDAL, PROJ INFO
---------------------
GEOS       : 3.12.1
GEOS lib   : None
GDAL       : 3.9.0
GDAL data dir: C:\miniforge\envs\data-science\Library\share\gdal\
PROJ       : 9.4.0
PROJ data dir: C:\miniforge\envs\data-science\Library\share\proj

PYTHON DEPENDENCIES
-------------------
geopandas  : 1.0.0
numpy      : 1.26.4
pandas     : 2.2.2
pyproj     : 3.6.1
shapely    : 2.0.4
pyogrio    : 0.8.0
geoalchemy2: None
geopy      : None
matplotlib : 3.8.4
mapclassify: 2.6.1
fiona      : 1.9.6
psycopg    : None
psycopg2   : None
pyarrow    : None

</details>

**Comment by martinfleis** [comment]
This is because the winding of the rings is always clockwise in this example. Normally, exterior is counter-clockwise while interiors are clockwise. If you normalize your polygon, it is plotted correctly.

```py
demo_gdf.normalize().plot(edgecolor='black')
```

However, given the polygon is valid, we should probably try to catch this on our side and normalize geometries prior plotting. The overhead will be negligible compared to the total cost of plotting.

**Comment by mathause** [comment]
See also https://github.com/shapely/shapely/pull/1933

**Comment by martinfleis** [comment]
Yeah, that is exactly the same issue and the same solution, just using older Python-based implementation. `normalize` shall be preferred now over `orient`.

**Comment by m-richards** [comment]
closed in #3483
